src/Tools/ExternalAccess/FSharp/Diagnostics/DiagnosticAnalyzerService.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections.Generic;
+using System.Composition;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.FSharp.Diagnostics
+{
+    [Shared]
+    [Export(typeof(IDiagnosticAnalyzerService))]
```

Comment:
❗️ Before this merges, we need a way to verify that this type will not be instantiated outside of F# scenarios.

Reply:
💡 It looks like this exports an instance of `IDiagnosticAnalyzerService`, which is also part of the ExternalAccess layer. This means we are safe on assembly loads, but it would help to use naming that makes it easier to understand this from the start:

```suggestion
    [Export(typeof(IFSharpDiagnosticAnalyzerService))]
```

Reply:
Hmm, actually I need to remove this as the shim for this specific service doesn't work yet. It will be in the second phase.

Reply:
Sorry, not this file.

 `DocumentDiagnosticAnalyzer.cs` needs to go away.

Reply:
Yea, I can change the name.